vtex-apps/product-highlights
# Add initial implementation

#### What problem is this solving?

Today there are three ways to add a highlight to a product:
1. By creating a product cluster and checking the flag to highlight
2. Creating a promotion
3. Creating a promotion with restrictions

This app handles those scenarios.

The app have a pretty similar API to [product-specifications](https://github.com/vtex-apps/product-specifications).

#### How to test it?

[Check the workspace](https://highlight--storecomponents.myvtex.com/apparel---accessories/clothing/?page=1)

Check the `blocks.json` added to the theme:
https://github.com/vtex-apps/store-theme/compare/feature/highlights

Added unit tests with 100% test coverage 😎 
![image](https://user-images.githubusercontent.com/284515/92051238-341bbe00-ed63-11ea-8744-32341e3b5b3a.png)



#### Screenshots or example usage:

![image](https://user-images.githubusercontent.com/284515/92049919-1437ca80-ed62-11ea-8ea5-ca0fb0a36dd8.png)

#### Describe alternatives you've considered, if any.

I've considered the API of [product-specification-badges](https://github.com/vtex-apps/product-specification-badges), but it's really complex with many props. The current approaches achieves the same things with a simpler API.

#### Related to / Depends on

n/a

#### How does this PR make you feel? [:link:](http://giphy.com/)



![](https://media.giphy.com/media/KBDGEAI1JHIftkUhjD/giphy.gif)
